zerver/openapi/zulip.yaml

```diff
@@ -510,6 +510,14 @@ paths:
               - type: string
               - type: integer
           example: 42
+        - name: is_web_public_query
+          in: query
+          description: If `true`, this is an unauthenticated request from a web public user.
+            If `false`, this is an authenticated request or the instance is not web public.
```

Comment:
We should explain the purpose here:
"""
Used when making unauthenticated requests to the Zulip API for organizations with web-public streams enabled.
This parameter exists mainly to ensure we still provide normal "Missing authentication" error messages for clients
not explicitly attempting to access web-public streams.
"""

Reply:
I'm also not sure this is the right interface for declaring this; maybe we should just be having `streams:web-public` be a narrow operator, if we think the right backend interface is to just add that operator when this option is passed anyway?

Reply:
Makes sense! If the narrow contains `streams:web-public`, we declare it as a `web_public_query`. We will redirect user to this narrow if the user is unauthenticated when accessing home.